src/main/scala/edu/cmu/cs/obsidian/parser/Parser.scala

```diff
@@ -601,12 +641,86 @@ object Parser extends Parsers {
         }
     }
 
+    private def parseContractType = {
+        parseId ~ opt(LBracketT() ~ repsep(parseType, CommaT()) ~ RBracketT()) ^^ {
+            case name ~ optParams =>
+                val params = optParams.map(_._1._2).getOrElse(List())
+                ContractType(name._1, params)
+        }
+    }
+
+    private def parseWhereImplements: Parser[Option[ContractType]] = {
+        WhereT() ~ parseId ~ ImplementsT() ~ parseType ^^ {
+            case _ ~ typeVar ~ _ ~ typeBound => typeBound match {
+                case np: NonPrimitiveType => Some(np.contractType)
+                case _ => failure("Contract cannot implement primitive type."); None
```

Comment:
If you give a parse error, the whole compilation process will stop. Can you give a type error in the typechecker instead?

Reply:
I could. The best way I can think of now is to make a new type of bound (e.g., `PrimitiveBound`), then in the typechecker if it encounters one of those it gives an errors and transforms it to the default `Top` bound. 

My initial thought is that it seems like a lot of complication for something that 1) people will probably almost never write and 2) has a pretty obvious fix.

Reply:
Yeah, that's the approach I came up with. It's fine to file an issue on this and leave the code as is.